python/sdk/merlin/model.py

```diff
@@ -1453,11 +1455,20 @@ def _wait_build_complete(self, logs):
 
 
 class PyFuncModel(PythonModel):
+    
     def load_context(self, context):
         self.initialize(context.artifacts)
         self._use_kwargs_infer = True
 
-    def predict(self, model_input, **kwargs):
+    def predict(self, context,  model_input):
+        extra_args = model_input.get(PYFUNC_EXTRA_ARGS_KEY, {})
+        input = model_input.get(PYFUNC_MODEL_INPUT_KEY, {})
+        if extra_args is not None and (type(extra_args) is dict) :
```

Comment:
Why do we need to add additional condition of ` and (type(extra_args) is dict)` ?